### DISCUSSION TITLE
Custom widget operation

### DISCUSSION THREAD

**Discussion by matthieugayon** [hecrj/iced#1438] [General]
Before stateless was the default widget API, it was easy to mutate a widget's state from the application's update method. 
Then stateless became the default api, and a new api for direct mutable access to the widgets state was introduced with [widget operations](https://github.com/iced-rs/iced/pull/1399), albeit in a "curated" fashion.

For my project I created a quite complex widget which requires to push and pull part of the widget state in a temporary memory, and for that matter I am using methods implemented on that state, which I used to call from the application update method. 
Right now, with the new default stateless API, I do not have a clear way of doing that, as it seems it is not possible to create custom widget operations. Is it ? 

Has anybody figured out a way to circumvent that limitation ? What would it take to expand the widget operation API to support custom user operations ?

**Comment by hecrj** [comment]
You can still store and manage widget state in your application, mutate it in your `update` logic, and provide an immutable reference to your widget in `view`.
